### PR TITLE
{,sec}boot: pass "key-name" to the FDE hooks

### PR DESCRIPTION
**Pull request by mvo5**
This is a followup to PR#9741 - we got asked that the FDE hooks
contains the key name. One of the existing hooks is using the
order in which the hooks are run to infere this informaton.

This commit adds the key-name to the key.

### CODE REVIEW

File: secboot/secboot.go [hunk 1]
```diff
@@ -56,7 +56,10 @@ func NewLoadChain(bf bootloader.BootFile, next ...*LoadChain) *LoadChain {
 type SealKeyRequest struct {
 	// The key to seal
 	Key EncryptionKey
-	// The path to store the sealed key file
+	// The key name, idential keys should have idential names
```
Comment (anonymouse64) [inline secboot/secboot.go:59]:
```suggestion
	// The key name; identical keys should have identical names
```

**Review by anonymouse64 — APPROVED** [review]
one typo, but otherwise LGTM, thanks for this

File: boot/seal.go [hunk 2]
```diff
@@ -129,10 +130,12 @@ func fallbackKeySealRequests(key, saveKey secboot.EncryptionKey) []secboot.SealK
 	return []secboot.SealKeyRequest{
 		{
 			Key:     key,
+			KeyName: "ubuntu-data-key",
```
Comment (bboozzoo) [inline boot/seal.go:133]: should these be ubuntu-recovery-data-key and ubuntu-recovery-save-key?

File: boot/seal.go [hunk 3]
```diff
@@ -120,6 +120,7 @@ func runKeySealRequests(key secboot.EncryptionKey) []secboot.SealKeyRequest {
...
```
Comment (pedronis) [inline boot/seal.go:123]: this should be just "ubuntu-data"

File: boot/seal.go (hunk 2 above)
Comment (pedronis) [reply boot/seal.go:133]: well, the key is the same, it depends what we mean by key-name,  that's why we left it out originally.  Anyway from my POV the name is the name of the bits in the key, same bits same name

Comment (pedronis) [reply boot/seal.go:133]: also here no "-key"

File: boot/seal.go [hunk 4]
```diff
@@ -129,10 +130,12 @@ func fallbackKeySealRequests(key, saveKey secboot.EncryptionKey) []secboot.SealK
...
 			KeyFile: filepath.Join(InitramfsSeedEncryptionKeyDir, "ubuntu-data.recovery.sealed-key"),
 		},
 		{
 			Key:     saveKey,
+			KeyName: "ubuntu-save-key",
```
Comment (pedronis) [inline boot/seal.go:138]: no "-key"

**Review by pedronis — CHANGES_REQUESTED** [review]
some tweaks are needed I think, to also match what we do on the unlock side

**Review by bboozzoo — APPROVED** [review]
LGTM

**Review by pedronis — APPROVED** [review]
thx, lgtm, as we discussed we need feedback whether this solves the original request, but the change in itself looks sound

**Comment by mvo5** [comment]
I asked Robert on mattermost and he confirmed this PR has what he needs to move to key-names instead of sequence numbers (that is what he is currently using).

**Pull request merged**